### PR TITLE
feat: add cta on complete page

### PR DESCRIPTION
**Pull request by catalinpit**

### CODE REVIEW

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import { PasswordInput } from '@documenso/ui/primitives/password-input';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+export type ClaimAccountProps = {
+  userName: string;
+  userEmail: string;
+  trigger?: React.ReactNode;
+};
+
+export const ZClaimAccountFormSchema = z
```
Comment (catalinpit) [inline apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:32]: Wanted to do import `ZSignUpFormSchema` from the "signup" component and do `ZSignUpFormSchema.omit`, but `refine` doesn't have the `omit` method.

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      name: userName ?? '',
+      email: userEmail,
+      password: '',
+    },
+    resolver: zodResolver(ZClaimAccountFormSchema),
+  });
+
+  const onFormSubmit = async ({ name, email, password }: TClaimAccountFormSchema) => {
+    try {
+      await signup({ name, email, password, signature: 'undefined' });
```
Comment (catalinpit) [inline apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:69]: for some reason, it doesn't allow me to pass `undefined` or `null`. only the `'undefined'` string.

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    password: ZPasswordSchema,
+  })
+  .refine(
+    (data) => {
+      const { name, email, password } = data;
+      return !password.includes(name) && !password.includes(email.split('@')[0]);
+    },
+    {
+      message: 'Password should not be common or based on personal information',
+      path: ['password'],
```
Comment (catalinpit) [inline apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:45]: this didn't work until I added the `path` property.

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx (hunk 2 above)
Comment (dguyen) [reply apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:69]: Isn't it because signature is required? And by doing this the signature of the user might be getting set to the literal string 'undefined'

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx (hunk 1 above)
Comment (Mythie) [reply apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:32]: It does not :(

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx (hunk 2 above)
Comment (Mythie) [reply apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:69]: Yup, we can loosen the `signup` method as part of this change to support these kind of quick actions.

File: apps/web/src/app/(signing)/sign/[token]/complete/page.tsx [hunk 4]
```diff
@@ -64,94 +66,107 @@ export default async function CompletedSigningPage({
...
-          )}
+    <div className="-mx-4 flex flex-col items-center overflow-x-hidden px-4 pt-24 md:-mx-8 md:px-8 lg:pt-36 xl:pt-44">
+      <div
+        className={`relative mt-6 flex w-full ${
+          isLoggedIn ? 'flex-col' : 'flex-col divide-y lg:flex-row lg:divide-x lg:divide-y-0'
+        } items-center justify-center`}
+      >
+        <div
+          className={`flex flex-col items-center ${isLoggedIn ? '' : 'mb-8 p-4 md:mb-0 md:p-12'}`}
+        >
```
Comment (dguyen) [inline apps/web/src/app/(signing)/sign/[token]/complete/page.tsx:77]: I recommend using cn

random examples

```
cn('base class stuff', isLoggedIn ? 'flex-col' : 'flex-col divide-y lg:flex-row lg:divide-x lg:divide-y-0')

cn('base class stuff', {
	'class stuff': isLoggedIn,
})

// Proper example
className={cn('my-2 flex items-center gap-2 font-normal', {
  'opacity-50': !value.includes(recipient.id),
})}
```

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+              )}
+            />
+            <FormField
+              name="email"
+              control={form.control}
+              render={({ field }) => (
+                <FormItem className="mt-4">
+                  <FormLabel>Email address</FormLabel>
+                  <FormControl>
+                    <Input {...field} value={userEmail} placeholder="Enter your email" />
```
Comment (dguyen) [inline apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:127]: Is setting `value={userEmail}` required here?

File: apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx (hunk 2 above)
Comment (catalinpit) [reply apps/web/src/app/(signing)/sign/[token]/complete/claim-account.tsx:69]: Yes @dguyen. I didn't want to change that initially, but I'll do it now as it seems that's the solution for now.

**Comment by catalinpit** [comment]
@coderabbit help

**Comment by catalinpit** [comment]
@coderabbitai help

**Comment by catalinpit** [comment]
@coderabbit can you summarize the changes in this repo for the last 2 weeks? Include the files that were changed the most, the number of issues and prs closed, and the number of lines of code added and removed. Thanks

**Comment by catalinpit** [comment]
@CodeRabbitAI can you summarize the changes in this repo for the last 2 weeks? Include the files that were changed the most, the number of issues and prs closed, and the number of lines of code added and removed. Thanks

**Comment by catalinpit** [comment]
@coderabbitai review

**Review by dguyen — APPROVED** [review]
Looks fine to me code wise

**Comment by dguyen** [comment]
Totally forgot about this, but can you hide this feature when NEXT_PUBLIC_DISABLE_SIGNUP is set to true?

File: packages/trpc/server/auth-router/router.ts [hunk 6]
```diff
@@ -38,8 +38,6 @@ export const authRouter = router({
 
       return user;
     } catch (err) {
-      console.log(err);
```
[the remark this answers is not in the document]
Comment (catalinpit) [reply packages/trpc/server/auth-router/router.ts:57]: Sure, but do we want to console errors?

File: apps/web/src/app/(signing)/sign/[token]/complete/page.tsx [hunk 7]
```diff
@@ -62,96 +68,112 @@ export default async function CompletedSigningPage({
 
   const sessionData = await getServerSession();
   const isLoggedIn = !!sessionData?.user;
+  const callToActionCheck = !isLoggedIn && NEXT_PUBLIC_DISABLE_SIGNUP?.toString() !== 'true';
```
Comment (catalinpit) [inline apps/web/src/app/(signing)/sign/[token]/complete/page.tsx:71]: `toString()` because people might use a boolean value in the `.env` file...

Comment (dguyen) [reply apps/web/src/app/(signing)/sign/[token]/complete/page.tsx:71]: I believe everything in an env file gets passed through as a string

**Review by dguyen — APPROVED** [review]
Looks fine to me besides possibly finding a more suitable name for `callToActionCheck`

**Comment by dguyen** [comment]
I was doing some testing on this, and noticed that there is some weird scrollbar stuff happening

image

**Comment by catalinpit** [comment]
> I was doing some testing on this, and noticed that there is some weird scrollbar stuff happening image
> 
> Inspecting the element shows that there's some weird height stuff happening

Are there 2 scrollbars? Cause I can't replicate it to have 2 scrollbars.

**Comment by catalinpit** [comment]
> I was doing some testing on this, and noticed that there is some weird scrollbar stuff happening image
> 
> Inspecting the element shows that there's some weird height stuff happening

Found the culprit. It's the `min-h-screen` from the `SigningLayout`.

**Comment by ElTimuro** [comment]
- Minor point, the copy should mention you get the document you signed in directly in the account

**Review by Mythie — DISMISSED** [review]

**Comment by dguyen** [comment]
Minor UI issues

The BG is cropped out

image

The top padding for unauthenticated layouts is gone

image

**Comment by catalinpit** [comment]
> Minor UI issues
> 
> 
> 
> The BG is cropped out
> 
> 
> 
> image
> 
> 
> 
> The top padding for unauthenticated layouts is gone
> 
> 
> 
> image
> 
> 

How do you keep finding these? 😆

I testes them locally a zillion times on different screens to make sure they look good and you still found something, haha. Will try to fix these! Thanks! 🙏

**Comment by dguyen** [comment]
> Regarding this "The top padding for unauthenticated layouts is gone", I removed it because there was an unnecessary scrolling. Should I add it back? Imo, it looks better like this.

As in it caused unnecessarily scrolling due to the general height of the page?

PDF lengths range in size, normally they're multi pages for signatures and such so there usually would be scrolling anyway

**Comment by catalinpit** [comment]
> > Regarding this "The top padding for unauthenticated layouts is gone", I removed it because there was an unnecessary scrolling. Should I add it back? Imo, it looks better like this.
> 
> 
> 
> As in it caused unnecessarily scrolling due to the general height of the page?
> 
> 
> 
> PDF lengths range in size, normally they're multi pages for signatures and such so there usually would be scrolling anyway

I'm writing these on my mobile, so I wasn't super explicit.

I know there would be scrolling because PDFs are of different sizes. 

"As in it caused unnecessarily scrolling due to the general height of the page?"

Yes. Even if you have a PDF with one page, you'll have to scroll because of excessive padding at the top.

**Comment by dguyen** [comment]
Got it 👍

**Review by dguyen — DISMISSED** [review]

**Review by dguyen — APPROVED** [review]
